File: FallDetection/Backend/src/main/java/com/example/demo/dto/Camera.java
package com.example.demo.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class Camera {
    private Long id;
    
    @JsonProperty("serial_number")
    private String serialNumber;
    
    @JsonProperty("friendly_name")
    private String friendlyName;
    
    @JsonProperty("room_name")
    private String roomName;
    
    @JsonProperty("is_online")
    private Boolean isOnline;
    
    private String model;
    private String version;
}

File: FallDetection/Frontend/lib/models/camera.dart
class Camera {
  final int id;
  final String serialNumber;
  final String friendlyName;
  final String roomName;
  final bool isOnline;
  final String model;
  final String version;

  Camera({
    required this.id,
    required this.serialNumber,
    required this.friendlyName,
    required this.roomName,
    required this.isOnline,
    required this.model,
    required this.version,
  });

  factory Camera.fromJson(Map<String, dynamic> json) {
    return Camera(
      id: json['id'],
      serialNumber: json['serial_number'],
      friendlyName: json['friendly_name'],
      roomName: json['room_name'],
      isOnline: json['is_online'],
      model: json['model'],
      version: json['version'],
    );
  }
}